core/src/main/kotlin/com/malinskiy/marathon/execution/queue/TestState.kt

```diff
@@ -1,24 +1,36 @@
 package com.malinskiy.marathon.execution.queue
 
-import com.malinskiy.marathon.device.DeviceInfo
-import com.malinskiy.marathon.execution.TestResult
 
+/**
+ * TestState accumulates all the information about a single test case for one pool
+ */
 sealed class TestState {
-    data class Added(val count: Int) : TestState()
+    data class Added(
+        val total: Int, 
+        val running: Int = 0,
+    ) : TestState()
+    
+    data class Passing(
+        val total: Int,
+        val running: Int,
+        val done: Int,
+    ) : TestState()
 
-    data class Executed(
-        val device: DeviceInfo,
-        val testResult: TestResult,
-        val count: Int
+    data class Failing(
```

Comment:
Right now it's not fully cleared difference between Failed and Failing.
Maybe we could use additional differentiation: `Terminal` States and `NotTerminal`

Reply:
My impression is that using adjectives ending with `ing` is quite common for this purpose in English i.e. https://buildkite.com/docs/pipelines/defining-steps#build-states